xgi/algorithms/centrality.py

```diff
@@ -307,3 +308,88 @@ def line_vector_centrality(H):
             vc[node].append(c_i[node])
 
     return vc
+
+
+def katz_centrality(H, index=False, cutoff=100):
+    """Returns the Katz-centrality vector of a non-empty hypergraph H.
+
+    The Katz-centrality measures the relative importance of a node by counting
+    how many distinct walks start from it. The longer the walk is the smaller
+    its contribution will be (attenuation factor `alpha`).
+    Initialy defined for graphs, the Katz-centrality is here generalized to
+    hypergraphs using the most basic definition of neighbors : two nodes that
+    share an hyperedge.
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        Hypergraph on which to compute the Kayz-centralities.
+    index : bool
+        If set to `True`, will return a dictionary mapping each vector index to a
+        node. Default value is `False`.
+    cutoff : int
+        Power at which to stop the serie A + alpha * A**2 + alpha**2 * A**3 + ..
+        Default value is 100.
+
+    Returns
+    -------
+    c : np.ndarray
+        Vector of the node centralities, sorted by the node indexes.
+    nodedict : dict
+        If index is set to True, nodedict will contain the nodes ids, keyed by
+        their indice in vector `c`.
+        Thus, `c[key]` will be the centrality of node `nodedict[key]`.
+
+    Raises
+    ------
+    XGIError
+        If the hypergraph is empty.
+
+    Notes
+    -----
+    [1] The Katz-centrality is defined as :
+        c = [(I - alpha.A^{t})^{-1} - I] • (1, 1, ..., 1)
+    Where A is the adjency matrix of the the (hyper)graph.
+    Since A^{t} = A for undirected graphs (our case), we have :
+        (I + A + alpha * A**2 + alpha**2 * A**3 + ...) * (I - alpha.A^{t})
+            = (I + A + alpha * A**2 + alpha**2 * A**3 + ...) * (I - alpha.A)
+            = (I + A + alpha * A**2 + alpha**2 * A**3 + ...) - A - alpha * A**2
+                - alpha**2 * A**3 - alpha**3 * A**4 - ...
+            = I
+    And (I - alpha.A^{t})^{-1} = I + A + alpha * A**2 + alpha**2 * A**3 + ...
+    Thus we can use the power serie to compute the Katz-centrality.
+    [2] The Katz-centrality of isolated nodes (no hyperedges contains them) is
+    zero. The Katz-centrality of an empty hypergraph is not defined.
+
+    References
+    ----------
+    See https://en.wikipedia.org/wiki/Katz_centrality#Alpha_centrality (visited
+    May 20 2023) for a clear definition of Katz centrality.
+    """
+
+    if index:
+        I, nodedict, _ = incidence_matrix(H, index=True)
+    else:
+        I = incidence_matrix(H, index=False)
+
+    N = len(H.nodes)
+    M = len(H.edges)
+    if N == 0:  # no nodes
+        raise XGIError("The Katz-centrality of an empty hypergraph is not defined.")
+    elif M == 0:
+        c = np.zeros(N)
+    else:  # there is at least one edge, both N and M are non-zero
+        A = I.dot(I.transpose()) - np.diag(np.sum(I, axis=1))
```

Comment:
There's a function for this called `clique_motif_matrix()`

Reply:
I put it in the code.
Why not `adjancy_matrix` ?

Reply:
You're right - `A = adjacency_matrix(weighted=True)` is the same thing. Totally fine whichever one you want to use.